src/Externs.cs

```diff
@@ -63,6 +63,53 @@ public partial class __default {
       return charseq.FromString(n.ToString());
     }
 
+    private static bool TryBigRationalToDouble(BigRational r, out Double d) {
+      // This is kind of ridiculous. We could certainly improve the BigRational
+      // class to make stuff like this easier. Conversion to and from `double`
+      // would probably be worth including.
+      string rStr = r.ToString();
```

Comment:
Are `num` and `den` private?  If they are less use reflection to get to them for now (https://stackoverflow.com/questions/3303126/how-to-get-the-value-of-private-field-using-reflection/3303182#3303182); then let's patch Dafny.  But we should not be parsing strings, I think

Reply:
Yeah, they are. We should update the C# runtime to make them accessible, but that'll need to wait for a Dafny release, so this is a stopgap. I hope it won't stick around for more than a few weeks.

Reply:
Why not just reflection in the meantime?

Reply:
It seems similarly awkward to me? I guess it's probably faster. Given that this is in place and works, I'd be more inclined to focus on making neither one necessary. :)

Reply:
It's simple and reliable (not pretty, but we know if works).  This string parsing scares me.  

Not a big deal, let's make a PR on he heels of this one to expose AsDouble.